lib/install/install.sh

```diff
@@ -193,6 +193,19 @@ bpkg_install () {
   return 0
 }
 
+bpkg_solve_uri() {
```

Comment:
Maybe `bpkg_solve_uri` could look for a user space function that can format a pathspec like this...

thinking out loud:

```sh
# in ~/.bpkgrc

bpkg_resolve_uri_pathspec () {
  # ...
}
```

in the `bpkg_solve_uri` function we could do:

```sh
bpkg_solve_uri() {
  if test -f bpkg_resolve_uri_pathspec; then
    bpkg_resolve_uri_pathspec $@
    return $?
  fi

  # ... rest of implementation
}
```

Reply:
Maybe using some configuration instead of code. Let's say that in `$HOME/.bpkg/urlmap` file with one regex that on matching gives the URI to be used. @jwerle Maybe your proposal fits better in bpkg. Just let me know

```
https://gitlab*:$user/$name/-/raw/$version
https://github*:$user/$name/$version
```

Reply:
I started looking into adding this feature a while back.  Essentially providing configurable repository sites.  The default would be GitHub, but we could add additional.  The bpkg.json could have a section for repositories to provide project specific dependencies.

On top of that, `bpkg.sh` provides a collection JSON and that would be a default, but others could setup their own (ie: internal teams and projects).  Collections (as they are now) could be updated to get the latest project repositories.  Subsequent project/system installs would use this list first, unless a dependency is specified with a FQDN.

I'm going to look into this PR further and provide additional feedback.

Reply:
It's OK to use [ShellSpec](https://shellspec.info/) for testing? Following https://github.com/bpkg/bpkg/issues/133

Reply:
Added external URL resolve tested with ShellSpec. Test execution : `shellspec --shell /bin/bash`